### PR TITLE
[TD-10456]<feature>taosdemo support schemaless

### PR DESCRIPTION
**Pull request by zhaoyanggh**

### CODE REVIEW

File: src/kit/taosdemo/taosdemo.c [hunk 1]
```diff
@@ -3795,6 +3795,9 @@ static int calcRowLen(SSuperTable*  superTbls) {
                 errorPrint2("get error data type : %s\n", dataType);
                 exit(EXIT_FAILURE);
         }
+        if( SML_IFACE == superTbls->iface) {
+            lenOfOneRow += 7;
```
Comment (sangshuduo) [inline src/kit/taosdemo/taosdemo.c]: magic number

File: src/kit/taosdemo/taosdemo.c [hunk 2]
```diff
@@ -10998,9 +11014,15 @@ static void* syncWriteProgressiveSml(threadInfo *pThreadInfo) {
     }
 
     if (stbInfo->lineProtocol == TSDB_SML_JSON_PROTOCOL) {
+        tmfree(pThreadInfo->lines[0]);
         tmfree(pThreadInfo->lines);
 free_json_progressive_sml:
```
Comment (sangshuduo) [inline src/kit/taosdemo/taosdemo.c:10988]: indent

File: src/kit/taosdemo/taosdemo.c [hunk 3]
```diff
@@ -9616,217 +9651,459 @@ static void* syncWriteInterlaceStmt(threadInfo *pThreadInfo, uint32_t interlaceR
...
+        dataLen += snprintf(sml + dataLen, length - dataLen,
+                            "id=%s%" PRIu64 "", stbInfo->childTblPrefix,
+                            tbSeq + pThreadInfo->start_table_from);
+    } else {
+        errorPrint2("[%d]: %s() LN%d, unsupport schemaless protocol (%d)\n", pThreadInfo->threadID, __func__,
+            __LINE__, stbInfo->lineProtocol);
+        return -1;
+    }
+    
         for (int j = 0; j < stbInfo->tagCount; j++) {
```
Comment (sangshuduo) [inline src/kit/taosdemo/taosdemo.c]: indent issue?

File: src/kit/taosdemo/taosdemo.c [hunk 4]
```diff
@@ -9616,217 +9651,459 @@ static void* syncWriteInterlaceStmt(threadInfo *pThreadInfo, uint32_t interlaceR
...
+    cJSON_AddItemToObject(record, "timestamp", ts);
+    cJSON* value = cJSON_CreateObject();
+    switch (stbInfo->columns[0].data_type) {
+        case TSDB_DATA_TYPE_BOOL:
+            cJSON_AddNumberToObject(value, "value", rand_bool());
+            cJSON_AddStringToObject(value, "type", "bool");
+            break;
+        case TSDB_DATA_TYPE_UTINYINT:
+        case TSDB_DATA_TYPE_TINYINT:
+            cJSON_AddNumberToObject(value, "value", rand_tinyint());
```
Comment (sangshuduo) [inline src/kit/taosdemo/taosdemo.c:10051]: should we support rand_utinyint(), rand_usmallint(), etc to test boundary more precisely?

**Review by sangshuduo — COMMENTED** [review]
Should we add new case to full test for CI?

**Comment by zhaoyanggh** [comment]
> Should we add new case to full test for CI?

yes, haoran is going to add some new cases to this branch.

File: src/kit/taosdemo/taosdemo.c (hunk 4 above)
Comment (zhaoyanggh) [reply src/kit/taosdemo/taosdemo.c:10051]: We don't support unsigned data type for schemaless JSON protocol, I can either change the unsigned data to invalid data type or keep this which deals with unsigned and signed data in the same way.

Comment (sangshuduo) [reply src/kit/taosdemo/taosdemo.c:10051]: let user know unsigned is invalid is better.

**Comment by sangshuduo** [comment]
> 
> 
> > Should we add new case to full test for CI?
> 
> yes, haoran is going to add some new cases to this branch.

OK

**Pull request closed without merging**